action/proto/proto.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+from typing import Optional
+
+import proto_schema_parser.parser as proto_parser
+from proto_schema_parser.ast import Enum, EnumValue
+
+
+class ProtoReader:
+    def __init__(self, workspace: Path, proto_path_str: Optional[str] = ""):
+        """Create a new ProtoReader instance.
+
+        Args:
+            workspace (Path): The path of the workspace.
+            proto_path_str (str, optional): The path to the proto files.
+        """
+        self.proto_code_dict = {}
+        if proto_path_str:
```

Comment:
What happens when it's empty? Does it even do anything then?

Also, Python convention is accepting a path as `proto_path: Union[str, Path]` and then `proto_path = Path(proto_path)` to ensure it's a `Path`.

Reply:
Yeah, my thinking was when we don't pass any path then just do nothing